tests/qxmppmixinvitation/tst_qxmppmixinvitation.cpp

```diff
@@ -0,0 +1,115 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ */
+
+#include "QXmppMixInvitation.h"
+
+#include <QDomDocument>
+#include <QObject>
+
+#include "util.h"
+
+class tst_QXmppMixInvitation : public QObject
+{
+    Q_OBJECT
+
+private slots:
+    void testInvitation();
+    void testIsInvitation();
+};
+
+void tst_QXmppMixInvitation::testInvitation()
+{
+    const QByteArray xml(
+        "<invitation xmlns=\"urn:xmpp:mix:misc:0\">"
+            "<inviter>hag66@shakespeare.example</inviter>"
+            "<invitee>cat@shakespeare.example</invitee>"
+            "<channel>coven@mix.shakespeare.example</channel>"
+            "<token>ABCDEF</token>"
+        "</invitation>"
+    );
+
+    QXmppMixInvitation invitation;
+    parsePacket(invitation, xml);
+
+    QCOMPARE(invitation.inviterJid(), QStringLiteral("hag66@shakespeare.example"));
+    QCOMPARE(invitation.inviteeJid(), QStringLiteral("cat@shakespeare.example"));
+    QCOMPARE(invitation.channelJid(), QStringLiteral("coven@mix.shakespeare.example"));
+    QCOMPARE(invitation.token(), QStringLiteral("ABCDEF"));
+
+    serializePacket(invitation, xml);
+
+    invitation.setInviterJid("hag66@shakespeare.example");
+    invitation.setInviteeJid("cat@shakespeare.example");
+    invitation.setChannelJid("coven@mix.shakespeare.example");
+    invitation.setToken("ABCDEF");
+
+    QCOMPARE(invitation.inviterJid(), QStringLiteral("hag66@shakespeare.example"));
+    QCOMPARE(invitation.inviteeJid(), QStringLiteral("cat@shakespeare.example"));
+    QCOMPARE(invitation.channelJid(), QStringLiteral("coven@mix.shakespeare.example"));
+    QCOMPARE(invitation.token(), QStringLiteral("ABCDEF"));
+}
+
+void tst_QXmppMixInvitation::testIsInvitation()
+{
+    QDomDocument doc;
+    QDomElement element;
+
+    const QByteArray correctInvitationWithNamespace(
+        "<invitation xmlns=\"urn:xmpp:mix:misc:0\">"
+            "<inviter>hag66@shakespeare.example</inviter>"
+            "<invitee>cat@shakespeare.example</invitee>"
+            "<channel>coven@mix.shakespeare.example</channel>"
+            "<token>ABCDEF</token>"
+        "</invitation>"
+    );
+    QCOMPARE(doc.setContent(correctInvitationWithNamespace, true), true);
+    element = doc.documentElement();
+    QVERIFY(QXmppMixInvitation::isMixInvitation(element));
+
+    const QByteArray correctInvitationWithoutNamespace(
+        "<invitation>"
+            "<inviter>hag66@shakespeare.example</inviter>"
+            "<invitee>cat@shakespeare.example</invitee>"
+            "<channel>coven@mix.shakespeare.example</channel>"
+            "<token>ABCDEF</token>"
+        "</invitation>"
+    );
+    QCOMPARE(doc.setContent(correctInvitationWithoutNamespace, true), true);
+    element = doc.documentElement();
+    QVERIFY(QXmppMixInvitation::isMixInvitation(element));
```

Comment:
This is outdated / should be false now. After that the test should also pass again.